### PR TITLE
Added IP2Location.io API as failover geolocation API

### PR DESCRIPTION
**Pull request by ip2location**

### CODE REVIEW

**Comment by Vlad55432** [comment]
not a dev, but is this better than stock?

File: Bloxstrap/Integrations/ActivityWatcher.cs [hunk 1]
```diff
@@ -301,17 +301,36 @@ public async Task<string> GetServerLocation()
...
+					if (string.IsNullOrEmpty(ipInfo.Country))
+						location = "?";
+					else if (ipInfo.City == ipInfo.Region)
+						location = $"{ipInfo.Region}, {ipInfo.Country}";
+					else
+						location = $"{ipInfo.City}, {ipInfo.Region}, {ipInfo.Country}";
+
+				}
+
+				if (!ActivityInGame)
```
Comment (axellse) [inline Bloxstrap/Integrations/ActivityWatcher.cs:333]: what is going on with the indentation here lol

**Review by axellse — CHANGES_REQUESTED** [review]
also two more things: 

1. Has ipinfo ever been down? otherwise this seems kinda unnecessary.
2. This "ip2location" api is a service you own, and searching for "ip location api" on google i couldn't find it, even after scrolling 5 pages. If a fallback service would be added it would make a lot more sense to use a highly ranked and popular one instead.
3.
bild
Its limited to 500 queries per day, then it costs money?? now it makes even less sense to add this, considering there are many other services doing it for completly free lol.

**Comment by Storm-930** [comment]
dude if you want to geolocate IPs you got angry IP scanner for that

**Comment by pizzaboxer** [comment]
i am in awe at the strategy here of e-stalking repos using competing geolocation apis but no thank you

**Comment by WaviestBalloon** [comment]
> i am in awe at the strategy here of e-stalking repos using competing geolocation apis but no thank you

And the fact they do this for multiple languages is crazy

**Comment by abdullahdevrel** [comment]
Hey folks,

I work for IPinfo. Thank you for your decision to continue using our service.

We respect the decision of this project's maintainers to choose which IP data provider they use, but please do let us know if there is any way we can do better. We have continuously made investments in supporting OSS projects, such as releasing our free full-accuracy database, community platform, community events etc.

I, too, regularly reach out to open-source projects, but I try to do so respectfully. However, I do not do it as a business strategy, as the data/API I pitch is completely free. We surely have an advantage over some of our competitors, and I am pretty confident in our data and the developer experience we provide. 

Just for context:

- IPinfo.io provides 1,000 requests per day without a token.
- IPinfo provides 50,000 requests per month with a free tier token.
- Universal access to a globally distributed load-balancing network regardless of access tier.
- Higher accuracy driven by latency-based IP geolocation methods. ([Blog post](https://ipinfo.io/blog/probe-network-how-we-make-sure-our-data-is-accurate/))
- Free HTTPS/SSL encryption
- Free IP to Country ASN database (daily updates, full accuracy, zero compromise)
- Bunch of free libraries, integrations, partnerships with platforms, online tools and developer tools
- An entire community platform dedicated to developers
- Great uptime hours and the capacity to support billions of requests each hour.


Just let me know if you have any feedback for us, and I will try my best to help.

Best Regards,
Abdullah (DevRel @ IPinfo.io)

**Pull request closed without merging**